eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs

```diff
@@ -215,17 +216,35 @@ isValidPlutusScript pv ps = withPlutusScript ps (isValidPlutus pv)
 -- Alonzo Plutus Purpose =======================================================
 
 newtype AsIndex ix it = AsIndex {unAsIndex :: ix}
-  deriving newtype (Eq, Ord, Show, NFData, NoThunks, EncCBOR, DecCBOR, Generic)
+  deriving stock (Show)
+  deriving newtype (Eq, Ord, NFData, NoThunks, EncCBOR, DecCBOR, Generic)
 
 newtype AsItem ix it = AsItem {unAsItem :: it}
-  deriving newtype (Eq, Ord, Show, NFData, NoThunks, EncCBOR, DecCBOR, Generic)
+  deriving stock (Show)
+  deriving newtype (Eq, Ord, NFData, NoThunks, EncCBOR, DecCBOR, Generic)
 
-instance ToJSON a => ToJSON (AsIndex a b) where
+data AsIxItem ix it = AsIxItem
```

Comment:
A little inconsistent with the `AsIndex`, which I presume refers to the same context? 
Is `AsIndexItem` too cumbersome?

Reply:
It has bothered me too, I was actually gonna rename `AsIndex` to `AsIx`, but that was much work. But, I just don't like `AsIndexItem`, because it just doesn't sound right. It would have to be `AsIndexedItem`, but that just too long, especially since it is always used as part of another type, eg: :grin: 
```haskell
AlonzoPlutusPurpose AsIndexedItem (AlonzoEra StandardCrypto)
```

Reply:
What do you think, should we rename `AsIndex` to `AsIx` for consistency?

I could create a tech dept ticket that we could address at some point later.

Reply:
I think `Ix` is more common in our codebase for types than `Index`  (except for MemoHashIndex..), but we have `TxIx` and `Data.Ix`.  
So I think what you propose, a ticket to change to `AsIx`  is better than what I said `AsIndexItem`, which indeed, is long, ungrammatical and inconsistent!  :D

Reply:
Agree. Created a ticket: https://github.com/IntersectMBO/cardano-ledger/issues/4036